### DISCUSSION TITLE
[Closed] AllowsMultipleExecutions missing from AsyncRelayCommand

### DISCUSSION THREAD

**Discussion by Orgbrat** [CommunityToolkit/Maui#131] [New Feature Discussions]
In Maui the maintainers are saying if you use MVVM then you should use CommunityToolkit.Mvvm by Microsoft.Toolkit which is perfectly fine. It does have everything that you would need from XCT ObjectModel. But the AsyncRelayCommand does not have a way to stop multiple executions like the old XCT toolkit did.  And that was really handy because almost 100% of my calls into XCT AsyncCommand had allowsMultipleExecutions set to false so the user could not double click buttons and stuff.  I really do miss that in the AsyncRelayCommand in the CommunityToolkit.Mvvm.  Not sure if this is where we need to talk about this but it is being left out of the new Maui CommunityToolkit so I thought we would start here and see what other people had to say about this.

Steve Graddy

**Comment by TheCodeTraveler** [comment]
Hi Steve! 

I recommend escalating this on the MVVM Toolkit: https://github.com/CommunityToolkit/WindowsCommunityToolkit

For now, the recommended work-around is to use [Xamarin.CommunityToolkit.MauiCompat](https://www.nuget.org/packages/Xamarin.CommunityToolkit.Markup.MauiCompat/), which contains the same `AsyncCommand` implementation as the Xamarin.CommunityToolkit and is compatible with .NET MAUI. 

You can find more information about `Xamarin.CommunityToolkit.MauiCompat` here: [https://devblogs.microsoft.com/xamarin/introducing-net-maui-compatibility-for-the-xamarin-community-toolkit/](https://devblogs.microsoft.com/xamarin/introducing-net-maui-compatibility-for-the-xamarin-community-toolkit/?WT.mc_id=mobile-0000-bramin)

**Comment by Orgbrat** [comment]
@brminnick  I am working on migrating an existing XF application to Maui but I am rewriting it to be 100% Maui so I have moved this request to the MVVM Toolkit repo with a new discussion at [#4302](https://github.com/CommunityToolkit/WindowsCommunityToolkit/discussions/4302).  Thanks for you help...

Steve

**Comment by Sergio0694** [comment]
Hi @Orgbrat, this is a perfectly reasonable feature request and in fact it was already on my to-do list. We had to pause work for a bit for the 7.1 release of the Windows Community Toolkit, and then to take the time to split off all the .NET packages into the new .NET Community Toolkit (which will have its own separate repo), so we didn't want to open any new PRs for the .NET packages there that might potentially "get stuck" during this transition. I already have this feature implemented and prototyped locally, and I plan to open a PR to merge this into the MVVM Toolkit as one of the first new features there. Once that's in, you'll be able to grab a preview package from our CI and start using it into your application. The API shape is pretty much the same as you suggested: there will be new constructor overloads taking a `bool` parameter to indicate whether the command is allowed to execute multiple invocations concurrently, and setting this will affect how `CanExecute` will behave (and consequently, how `Execute` will behave as well). Hope this helps! 😄

**Comment by CliffAgius** [comment]
Closed as Answered.
